ext/weblocks/00_weblocks.js

```diff
@@ -0,0 +1,223 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+// referenced https://github.com/metarhia/web-locks/blob/master/web-locks.js
+
+/// <reference path="../../core/internal.d.ts" />
+/// <reference path="../../core/lib.deno_core.d.ts" />
+/// <reference path="../webidl/internal.d.ts" />
+
+"use strict";
+
+((window) => {
+  const webidl = window.__bootstrap.webidl;
+  const {
+    Promise,
+    Int32Array,
+    Map,
+    Set,
+  } = window.__bootstrap.primordials;
+
+  const LockMode = ["shared", "exclusive"];
+  const LockState = {
+    LOCKED: 0,
+    UNLOCKED: 1,
+  };
+
+  class Lock {
+    queue = [];
+    owner = false;
+    trying = false;
+    name;
+    mode;
+    constructor(name, mode = LockMode[0], buffer = null) {
```

Comment:
should be a illegal constructor and use `webidl.createBranded` and assign that way.